src/lstate.js

```diff
@@ -69,14 +69,23 @@ class global_State {
 
     constructor(L) {
         this.mainthread = L;
-        this.strt = null; // TODO: string hash table
+        this.strt = new Map();
```

Comment:
I wouldn't implement a global string table at all: I'd just do it at the point of comparison/set-in-table.

Additionally I was contemplating having two separate string implementations: for "normal" strings (i.e. 7-bit strings), just use normal JS strings. If they go multibyte, use a `UInt8Array`